src/tools/schema2kotlin.ts

```diff
@@ -116,6 +135,25 @@ abstract class Abstract${particleName} : ${this.opts.wasm ? 'WasmParticleImpl' :
 `;
   }
 
+  private getQueryType(connection: HandleConnectionSpec): string {
+    if (!(connection.type instanceof CollectionType)) {
+      return null;
+    }
+    const handleType = connection.type.getContainedType();
+    if (!(handleType instanceof EntityType)) {
```

Comment:
What about a collection of references? 

Maybe `type.getEntitySchema()` is safer.

Reply:
I don't think we are planning to support querying over collections of references (just yet).

I don't mind adding support for it, but using `type.getEntitySchema()` would only be part of the fix, so I'd prefer to return an error for now (i.e. the query code will have to handle dereferencing).